crates/re_types/src/archetypes/mesh3d.rs

```diff
@@ -0,0 +1,543 @@
+    once_cell::sync::Lazy::new(|| {
+        [
+            "rerun.components.Position3D".into(),
+            "rerun.components.Mesh3DIndicator".into(),
+            "rerun.components.MeshProperties".into(),
+            "rerun.components.ClassId".into(),
+            "rerun.components.Color".into(),
+            "rerun.components.InstanceKey".into(),
+            "rerun.components.Material".into(),
+            "rerun.components.Vector3D".into(),
+        ]
+    });
+
+impl Mesh3D {
+    pub const NUM_COMPONENTS: usize = 8usize;
+}
+
+/// Indicator component for the [`Mesh3D`] [`crate::Archetype`]
+pub type Mesh3DIndicator = crate::GenericIndicatorComponent<Mesh3D>;
+
+impl crate::Archetype for Mesh3D {
+    type Indicator = Mesh3DIndicator;
+
+    #[inline]
+    fn name() -> crate::ArchetypeName {
+        "rerun.archetypes.Mesh3D".into()
+    }
+
+    #[inline]
+    fn required_components() -> ::std::borrow::Cow<'static, [crate::ComponentName]> {
+        REQUIRED_COMPONENTS.as_slice().into()
+    }
+
+    #[inline]
+    fn recommended_components() -> ::std::borrow::Cow<'static, [crate::ComponentName]> {
+        RECOMMENDED_COMPONENTS.as_slice().into()
+    }
+
+    #[inline]
+    fn optional_components() -> ::std::borrow::Cow<'static, [crate::ComponentName]> {
+        OPTIONAL_COMPONENTS.as_slice().into()
+    }
+
+    #[inline]
+    fn all_components() -> ::std::borrow::Cow<'static, [crate::ComponentName]> {
+        ALL_COMPONENTS.as_slice().into()
+    }
+
+    #[inline]
+    fn num_instances(&self) -> usize {
+        self.vertex_positions.len()
+    }
+
+    fn as_component_batches(&self) -> Vec<crate::MaybeOwnedComponentBatch<'_>> {
+        [
+            Some(Self::Indicator::batch(self.num_instances() as _).into()),
+            Some((&self.vertex_positions as &dyn crate::ComponentBatch).into()),
+            self.mesh_properties
+                .as_ref()
+                .map(|comp| (comp as &dyn crate::ComponentBatch).into()),
+            self.vertex_normals
+                .as_ref()
+                .map(|comp_batch| (comp_batch as &dyn crate::ComponentBatch).into()),
+            self.vertex_colors
+                .as_ref()
+                .map(|comp_batch| (comp_batch as &dyn crate::ComponentBatch).into()),
+            self.mesh_material
+                .as_ref()
+                .map(|comp| (comp as &dyn crate::ComponentBatch).into()),
+            self.class_ids
+                .as_ref()
+                .map(|comp_batch| (comp_batch as &dyn crate::ComponentBatch).into()),
+            self.instance_keys
+                .as_ref()
+                .map(|comp_batch| (comp_batch as &dyn crate::ComponentBatch).into()),
+        ]
+        .into_iter()
+        .flatten()
+        .collect()
+    }
+
+    #[inline]
+    fn try_to_arrow(
+        &self,
+    ) -> crate::SerializationResult<
+        Vec<(::arrow2::datatypes::Field, Box<dyn ::arrow2::array::Array>)>,
+    > {
```

Comment:
Now that we have `as_component_batches`, can't `try_to_arrow` just call `self.as_component_batches` and then call `try_to_arrow()` on each component batch? A lot less code needs generating. I guess we get an additional `dyn` call per component, but that seems worth it

Reply:
- https://github.com/rerun-io/rerun/issues/3386